storage/google/cloud/storage/blob.py

```diff
@@ -468,6 +468,11 @@ def generate_signed_url(
         else:
             helper = generate_signed_url_v4
 
+        if self._encryption_key is not None:
```

Comment:
this should only be done if v4.

Reply:
Per chat, we still need to add the `X-Goog-Encryption-Algorithm` header for V2 (but not the key or hash, oddly enough).  See: https://cloud.google.com/storage/docs/access-control/signed-urls-v2#about-canonical-extension-headers